skbio/metadata/_testing.py

```diff
@@ -922,6 +922,15 @@ def test_constructor_invalid(self):
                                     'object.'):
             self._interval_metadata_constructor_(0, '')
 
+    def test_constructor_interval_metadata_upper_bound_is_none(self):
```

Comment:
Please add these test cases:

- successfully constructing from a non-empty `IntervalMetadata` object with `upper_bound=None`
- failing to construct from a non-empty `IntervalMetadata` object with `upper_bound=None` and coordinates that fall outside the mixin's coordinate space

Please also add similar test cases for the mixin setter (including `test_constructor_interval_metadata_upper_bound_is_none`, that would be 3 test cases for the setter).